lib/server.js

```diff
@@ -161,40 +161,40 @@ CoAPServer.prototype.listen = function(port, address, done) {
     address = null
   }
 
-  if (this._sock && done)
-    done(new Error('Already listening'))
-  else if (this._sock)
-    throw new Error('Already listening')
-
   if (address && net.isIPv6(address))
     this._options.type = 'udp6'
 
   if (!this._options.type)
     this._options.type = 'udp4'
 
-  this._sock = dgram.createSocket(this._options.type, handleRequest(this))
-
-  this._sock.on('error', function(error) {
-    that.emit('error', error)
-  })
-
-  this._sock.bind(port, address || null, done || null)
-  this._port = port
-  this._address = address
+  if (this._sock && done)
+    done(new Error('Already listening'))
+  else if (this._sock)
+    throw new Error('Already listening')
+  else {
```

Comment:
there is no need to have an else here, just do an early return from the previous block